src/lib/chat/openai.ts

```diff
@@ -0,0 +1,55 @@
+import OpenAI from 'openai';
+import type { ChatCompletionMessageParam } from 'openai/resources/index.mjs';
+import { get } from 'svelte/store';
+
+import { settingsStore } from '../localStorage';
+import type { ChatRequest, ChatStrategy, Model } from './index';
+
+export class OpenAIStrategy implements ChatStrategy {
+	private openai: OpenAI;
+
+	constructor() {
+		const settings = get(settingsStore);
+
+		this.openai = new OpenAI({
+			baseURL: settings.openaiServer,
+			apiKey: settings.openaiApiKey || '',
```

Comment:
There is an issue with this approach, we shouldn't instantiate a `new OpenAI()` object without an `apiKey`.

Looks like `|| ''` was added to satisfy the TS linter, but his code is currently causing `401` responses where there is no `apiKey` defined in Settings (which is the default behavior):

![image](https://github.com/user-attachments/assets/c03bdfad-81d0-47c4-b648-2977d9f00bfb)

Reply:
Likewise we also shouldn't instantiate `new OpenAI` without a `openaiServer`

Reply:
Maybe instead of reading `openAiApiKey` and `openAiServer` from the store, maybe they should be required arguments, for example: `new OpenAIStrategy(server: string, apiKey: string)`.